BLSQ/iaso
# POLIO-768 override step email notification

The budget tool's override step didn't send email notifications

Related JIRA tickets : POLIO-768

## Self proof reading checklist


## Changes

- Modify `base_budget_email.html` to change the wording when the step is "override"
- Change the `MailTemplate`' `render_for_step`method to be able to handle override steps ie: change the way the `node_key` is determined and pass the `override`boolean to the template's context
- Update override step's serializer to send email and handle node_key format received

## How to test

Configure your local setup to use SMTP bucket : https://wiki.bluesquare.org/en/Dev-team/product-management/iaso/email

Go to polio >budget as super user

Use the override button to set the state of the budget

Go to smtp bucket to see that the email is correctly sent. (NOTE: the buttons won't work if you don't change the default example.com domain: https://github.com/BLSQ/iaso#e-mails)

## Print screen / video

https://user-images.githubusercontent.com/38907762/214082281-257d91d4-7d96-4c25-9d1c-5ab437e4b312.mov

## Notes

@olethanh pls check it out as I had to workaround a few quirks, so I'm not sure it's the cleanest code ever
